logdna/meta.go

```diff
@@ -0,0 +1,89 @@
+package logdna
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type OrgType string
+type TerraformType string
+type TerraformInfo struct {
+	name          string
+	orgType       OrgType
+	terraformType TerraformType
+	schema        *schema.Resource
+}
+
+const (
+	OrgTypeRegular    OrgType = "regular"
+	OrgTypeEnterprise OrgType = "enterprise"
+)
+
+const (
+	TerraformTypeResource   TerraformType = "resource"
+	TerraformTypeDataSource TerraformType = "data source"
+)
+
+var terraformRegistry []TerraformInfo
+
+func registerTerraform(info TerraformInfo) *TerraformInfo {
+	terraformRegistry = append(terraformRegistry, info)
+	infoPt := &terraformRegistry[len(terraformRegistry)-1]
+
+	if infoPt.schema.CreateContext != nil {
+		infoPt.schema.CreateContext = buildTerraformFunc(infoPt.schema.CreateContext, infoPt)
+	}
+	if infoPt.schema.ReadContext != nil {
+		infoPt.schema.ReadContext = buildTerraformFunc(infoPt.schema.ReadContext, infoPt)
+	}
+	if infoPt.schema.UpdateContext != nil {
+		infoPt.schema.UpdateContext = buildTerraformFunc(infoPt.schema.UpdateContext, infoPt)
+	}
+	if infoPt.schema.DeleteContext != nil {
+		infoPt.schema.DeleteContext = buildTerraformFunc(infoPt.schema.DeleteContext, infoPt)
+	}
+
+	return infoPt
+}
+
+func filterRegistry(terraformType TerraformType) []TerraformInfo {
+	newSlice := []TerraformInfo{}
+
+	for _, info := range terraformRegistry {
+		if info.terraformType == terraformType {
+			newSlice = append(newSlice, info)
+		}
+	}
+
+	return newSlice
+}
+
+func buildSchemaMap(a []TerraformInfo) map[string]*schema.Resource {
+	m := make(map[string]*schema.Resource)
+
+	for _, e := range a {
+		m[e.name] = e.schema
+	}
+
+	return m
+}
```

Comment:
```suggestion
func buildSchemaMap(tfInfoList []TerraformInfo) map[string]*schema.Resource {
	schemaMap := make(map[string]*schema.Resource)

	for _, info := range tfInfoList {
		schemaMap[info.name] = info.schema
	}

	return schemaMap
}
```

Mainly, the `a` doesn't impart any information to the reader. Renaming the `m` and `e` variables would also really help the next person to look at this code.